file_ops.py

```diff
@@ -176,3 +194,17 @@ def _handle_duplicated_file_name(file_name, file_name_used_dict, split_ext=True)
         file_name_used_dict[file_name] = 0
 
     return file_name_new
+
+
+def _hash_string(_str):
+        hash_object = hashlib.md5(_str.encode())
+        return hash_object.hexdigest()
+
+
+def _get_cached_file(url, base_dir=CACHED_FILE_DIR):
+    hashkey = _hash_string(url)
```

Comment:
let's start cleanup refactoring to stop using private names for functions. get_cached_file() not _get_cached_file()